samples/bluetooth/peripheral_hr/src/main.c

```diff
@@ -136,14 +132,73 @@ int main(void)
 		return 0;
 	}
 
-	bt_ready();
+	printk("Bluetooth initialized\n");
 
 	bt_conn_auth_cb_register(&auth_cb_display);
 
 	bt_hrs_cb_register(&hrs_cb);
-	/* Implement notification. At the moment there is no suitable way
-	 * of starting delayed work so we do it here
-	 */
+
+#if !defined(CONFIG_BT_EXT_ADV)
+	printk("Starting Legacy Advertising (connectable and scannable)\n");
+	err = bt_le_adv_start(BT_LE_ADV_CONN_NAME, ad, ARRAY_SIZE(ad), NULL, 0);
+	if (err) {
+		printk("Advertising failed to start (err %d)\n", err);
+		return 0;
+	}
+
+#else /* CONFIG_BT_EXT_ADV */
+	struct bt_le_adv_param adv_param = {
+		.id = BT_ID_DEFAULT,
+		.sid = 0U,
+		.secondary_max_skip = 0U,
+		.options = (BT_LE_ADV_OPT_EXT_ADV |
+			    BT_LE_ADV_OPT_CONNECTABLE |
+			    BT_LE_ADV_OPT_CODED |
+			    BT_LE_ADV_OPT_USE_NAME),
+		.interval_min = BT_GAP_ADV_FAST_INT_MIN_2,
+		.interval_max = BT_GAP_ADV_FAST_INT_MAX_2,
+		.peer = NULL,
+	};
```

Comment:
To me it makes more sense to use the same parameters, only add the `BT_LE_ADV_OPT_EXT_ADV` for the extended parameters. 

That way we can also use the same restart functionality for both configs.

Reply:
Will keep as-is as this sample shows the use of two different API parameters, use of public define versus local parameter variable.